svg.js.d.ts

```diff
@@ -144,46 +614,421 @@ declare module "@svgdotjs/svg.js" {
+        node: HTMLElement | SVGElement;
+        type: string;
 
-    class Color implements ColorLike {
-        r: number;
-        g: number;
-        b: number;
         constructor();
-        constructor(color: ColorAlias);
+        constructor(node: HTMLElement, attr?: Object);
+        constructor(id: { id: string });
+        add(element: Element, i?: number): this;
+        addTo(parent: Dom | HTMLElement | string): this
+        children(): List<Element>;
+        clear(): this;
+        clone(): this;
+        each(block: (index: number, children: Element[]) => void, deep?: boolean): this;
+        element(element: string, inherit?: Object): this;
+        first(): Element;
+        get(i: number): Element;
+        getEventHolder(): LinkedHTMLElement;
+        getEventTarget(): LinkedHTMLElement;
+        has(element: Element): boolean;
+        id(): string
+        id(id: string): this
+        index(element: Element): number;
+        last(): Element;
+        matches(selector: string): boolean;
+        /**
+         * Get the parent of current element. The input query can be given with string, object type or none (undefined).
+         * The input is vary based on the implement in hierarchy of SVG.JS element or dom.
+         * 1. If The input is a string, the string value must be a valid HTML element tag name or svg tag name. e.g "svg" or "g" or "div"
+         * 2. If the given input is an object type then only SVG.JS object type is accept. e.g Dom, Svg or G
+         * 3. if the given input query is undefined then the element will return the closest parent in Dom hierarchy
+         *
+         * For more information see ParentQueryMapping.
+         * @param type can be either string, object type or undefined.
+         */
+        parent<T extends this>(type?: ParentQueryMapping<T>): ParentQueryResultMapping<T>;
+        put(element: Element, i?: number): Element;
+        /**
+         * Put the element into the given parent element. The input parent element can be vary base on the class implementation.
+         * 1. If the current class is a Dom then parent input is only accept a valid HTML element or a valid string id of HTML element
+         * 2. If the current class is an Svg then parent input can only Dom, Svg, G, A, a valid HTML element and a valid string id of HTML element
+         *
+         * For more information see PutInMapping.
+         * @param parent an object of SVG.JS Dom or implement container or a string id or a valid HTML element object.
+         */
+        putIn<T extends this>(parent: PutInMapping<T>): PutInResultMapping<T>;
+
+        remove(): this;
+        removeElement(element: Element): this;
+        replace<T extends Dom>(element: T): T;
+        round(precision?: number, map?: string[]): this
+        svg(): string;
+        svg(a: boolean, outer?: boolean): string
+        svg(a: null | Function, outer?: boolean): string
+        svg(a: string, outer?: boolean): string

```

Comment:
This is an import not an export. It returns the element itself (or in case outer=true it would return the parent)

Reply:
should it return an Svg?

Reply:
It returns `this` for outer=false and `Element` for outer=true. I am not sure if you can type that with Typescript.
But it will always return an `Element`. Btw: If you want we can also chat on gitter :)

Reply:
gitter sounds like a better choice. How do I do that?

Reply:
I added this logic

Reply:
You can find our gitter channel here: https://gitter.im/svgdotjs/svg.js

You can login with github :)